redux/actions/userDetailsActions.ts

```diff
@@ -0,0 +1,40 @@
+import { ipClient } from '../../services/ip';
+import { googlePlacesApiClient } from '../../services/googlePlaces';
+import { AnyAction } from 'redux';
+import { GeocodingResponse } from 'google__maps';
+
+export const getUserCoordinates = (): AnyAction => {
+  const promise = ipClient.getUserLocation().then((response) => {
+    const { lat, lon } = response.data;
+    return { latitude: lat, longitude: lon };
+  });
+  return {
+    type: 'GET_USER_COORDINATES',
+    payload: promise,
+  };
+};
+
+export interface Coordinates {
+  latitude: number;
+  longitude: number;
+}
+
+export const getUserLocation = ({ latitude, longitude }: Coordinates): AnyAction => {
+  const promise = googlePlacesApiClient.getLocationDetails({ lat: latitude, lon: longitude }).then((response) => {
+    const { results } = (response as unknown) as GeocodingResponse;
+    const locationDict = results
```

Comment:
Define the callbacks outside of this function and pass them in for better readability.